cql3/describe_processor.cc

```diff
@@ -0,0 +1,241 @@
+#include "cql3/describe_processor.hh"
+#include "cql3/column_identifier.hh"
+#include "cql3/column_specification.hh"
+#include "cql3/result_set.hh"
+#include "cql3/util.hh"
+#include "data_dictionary/keyspace_metadata.hh"
+#include "exceptions/exceptions.hh"
+#include <boost/range/algorithm/sort.hpp>
+#include <boost/range/adaptor/transformed.hpp>
+#include <memory>
+
+namespace cql3 {
+
+static const sstring KS = "system";
```

Comment:
KS for referring to system_keyspace, CF to specify column_family name for metadata result (copied from Cassandra)

Reply:
I know what the words "KS" and "CF" refer to, what I didn't understand is what they are used for... Does this feature add another system table to the database, and if so, why?

Reply:
This PR adds 2 virtual tables `describe_table` and `describe_cluster` (probably the first name could be a bit better). 
I'm using KS to find schema of those tables and CF is used when I'm manually creating result metadata (only in `DESC CLUSTER`)

I have to create it manually there because the `cql_statement::get_result_metadata` has signature: `seastar::shared_ptr<const metadata> get_result_metadata() const` and cluster description depends on `client_state` (whether the keyspace is set to user keyspace, then the `token range to endpoint` column is added). In other cases, there is predefined selection, from which the result metadata is taken.